test/Mono.Linker.Tests.Cases/Warnings/WarningSuppression/DetectRedundantSuppressionsInMembersAndTypes.cs

```diff
@@ -0,0 +1,179 @@
+			[UnconditionalSuppressMessage ("Test", "IL2071")]
+			public static void Test ()
+			{
+				DetectRedundantSuppressionsInMembersAndTypes.TrimmerCompatibleMethod ();
+			}
+		}
+
+		public class RedundantSuppressionOnLocalMethod
+		{
+			public static void Test ()
+			{
+				[ExpectedWarning ("IL2121", "IL2071", ProducedBy = ProducedBy.Trimmer)]
+				[UnconditionalSuppressMessage ("Test", "IL2071")]
+				void LocalMethod ()
+				{
+					DetectRedundantSuppressionsInMembersAndTypes.TrimmerCompatibleMethod ();
+				}
+
+				LocalMethod ();
+			}
+		}
+
+		public class RedundantSuppressionOnNestedType
+		{
+			public static void Test ()
+			{
+				NestedType.TrimmerCompatibleMethod ();
+			}
+
+			[ExpectedWarning ("IL2121", "IL2071", ProducedBy = ProducedBy.Trimmer)]
+			[UnconditionalSuppressMessage ("Test", "IL2071")]
+			public class NestedType
+			{
+				public static void TrimmerCompatibleMethod ()
+				{
+					DetectRedundantSuppressionsInMembersAndTypes.TrimmerCompatibleMethod ();
+				}
+			}
+		}
+
+		public class RedundantSuppressionOnProperty
+		{
+			public static void Test ()
+			{
+				var property = TrimmerCompatibleProperty;
+			}
+
+			public static string TrimmerCompatibleProperty {
+				[ExpectedWarning ("IL2121", "IL2071", ProducedBy = ProducedBy.Trimmer)]
+				[UnconditionalSuppressMessage ("Test", "IL2071")]
+				get {
+					return DetectRedundantSuppressionsInMembersAndTypes.TrimmerCompatibleMethod ();
+				}
+			}
+		}
+
+		[ExpectedWarning ("IL2121", "IL2072", ProducedBy = ProducedBy.Trimmer)]
+		[UnconditionalSuppressMessage ("Test", "IL2072")]
+		[ExpectedWarning ("IL2121", "IL2071", ProducedBy = ProducedBy.Trimmer)]
+		[UnconditionalSuppressMessage ("Test", "IL2071")]
+		public class MultipleRedundantSuppressions
+		{
+			[ExpectedWarning ("IL2121", "IL2072", ProducedBy = ProducedBy.Trimmer)]
+			[UnconditionalSuppressMessage ("Test", "IL2072")]
+			[ExpectedWarning ("IL2121", "IL2071", ProducedBy = ProducedBy.Trimmer)]
+			[UnconditionalSuppressMessage ("Test", "IL2071")]
+			public static void Test ()
+			{
+				DetectRedundantSuppressionsInMembersAndTypes.TrimmerCompatibleMethod ();
+			}
+		}
+
+		public class RedundantAndUsedSuppressions
+		{
+			[ExpectedWarning ("IL2121", "IL2071", ProducedBy = ProducedBy.Trimmer)]
+			[UnconditionalSuppressMessage ("Test", "IL2071")]
+			[UnconditionalSuppressMessage ("Test", "IL2072")]
+			public static void Test ()
+			{
+				Expression.Call (DetectRedundantSuppressionsInMembersAndTypes.TriggerUnrecognizedPattern (), "", Type.EmptyTypes);
+			}
+		}
+
+		public class DoNotReportNonLinkerSuppressions
+		{
+			[UnconditionalSuppressMessage ("Test", "IL3052")]
+			public static void Test ()
+			{
+				DetectRedundantSuppressionsInMembersAndTypes.TrimmerCompatibleMethod ();
+			}
+		}
+
+		// Methods with unreachable bodies are not processed https://github.com/dotnet/linker/issues/2921
+		public class DoNotReportSuppressionsOnMethodsConvertedToThrow
```

Comment:
The issue referenced here is only about feature switches - which is not the case tested here.
The case tested here basically "works" as expected (per our discussion).
I would probably just add a comment describing the intended behavior (it's not super clear from the test code)

Reply:
Fixed the issue link and added some comments.